contrib/pcap2emulation.py

```diff
@@ -0,0 +1,349 @@
+#!/usr/bin/python3
+# pylint: disable=invalid-name,missing-docstring
+#
+# Copyright (C) 2023 Collabora Ltd
+# Author: Frédéric Danis <frederic.danis@collabora.com>
+#
+# SPDX-License-Identifier: LGPL-2.1+
+
+import argparse
+import base64
+import json
+import os
+import subprocess
+import sys
+
+URB_INTERRUPT = 1
+URB_CONTROL = 2
+URB_BULK = 3
+
+DESCRIPTOR_DEVICE = "1"
+DESCRIPTOR_CONFIGURATION = "2"
+DESCRIPTOR_STRING = "3"
+DESCRIPTOR_INTERFACE = "4"
+
+
+def get_int(str):
+    if str[:2] == "0x":
+        return int(str, 16)
+    return int(str)
+
+
+class Pcap2Emulation:
+    def __init__(self, prefix, device_ids):
+        self.device = None
+        self.prefix = prefix
```

Comment:
i'd feel a lot more warm and fuzzy if these had mypy annotations -- not a blocker by any means, but it's certainly saved me quite a few headaches when I wrote the LVFS...